src/olympia/devhub/tests/test_views.py

```diff
@@ -1592,15 +1592,56 @@ def test_welcome_email_for_newbies(self, send_welcome_email_mock):
         send_welcome_email_mock.assert_called_with(
             self.addon.id, ['del@icio.us'], context)
 
+    @mock.patch.object(settings, 'SITE_URL', 'http://b.ro')
+    @mock.patch('olympia.devhub.tasks.send_welcome_email.delay')
+    def test_welcome_email_first_listed_addon(self, send_welcome_email_mock):
+        new_addon = addon_factory(is_listed=False)
+        new_addon.versions.update(channel=amo.RELEASE_CHANNEL_UNLISTED)
```

Comment:
I actually did in a different branch I'll submit next, didn't want to deal with the conflicts later :)